File: algorithms/sorting-algos/selection-sort.py
def selection_sort(nums):
    for i in range(0, len(nums)):
        lowestIndex = i
        for j in range(i + 1, len(nums)):
            if (nums[j] < nums[lowestIndex]):
                lowestIndex = j
        (nums[i], nums[lowestIndex]) = (nums[lowestIndex], nums[i])

numbers = [12, 19, 2, 17, 3, 8, 19, 20, 18, 1]

selection_sort(numbers)

print(numbers)

File: algorithms/sorting-algos/selection-sort.js
function selectionSort(nums) {
    for (i = 0; i < nums.length - 1; i++) {
        let smallestIndex = i
        for (j = i + 1; j < nums.length; j++) {
            if (nums[j] < nums[smallestIndex]) {
                smallestIndex = j
            }
        }
        if (smallestIndex !== i) {
            [nums[i], nums[smallestIndex]] = [nums[smallestIndex], nums[i]]
        }
    }
} 

let numbers = [12, 19, 2, 17, 3, 8, 19, 20, 18, 1]

selectionSort(numbers) 